bindings/experimental/distrdf/python/DistRDF/Proxy.py

```diff
@@ -203,3 +203,24 @@ def _(operation: Operation.Action, node: Node) -> ActionProxy:
 def _(operation: Operation.InstantAction, node: Node) -> Any:
     execute_graph(node)
     return node.value
+
+
+@get_proxy_for.register
+def _(operation: Operation.Snapshot, node: Node) -> Union[ActionProxy, Any]:
+    if len(operation.args) == 4:
+        # An RSnapshotOptions instance was passed as fourth argument
+        if operation.args[3].fLazy:
+            return ActionProxy(node)
```

Comment:
Can we invoke the `Action` overload of `get_proxy_for` from here? If we realize this should behave as an action, it's better to rely on the `Action`'s implementation of `get_proxy_for` instead of repeating the code here.

Reply:
Mmm good point I'll try to dispatch the right functions thanks!